app/src/lib/stats/stats-store.ts

```diff
@@ -1525,12 +1543,149 @@ export class StatsStore implements IStatsStore {
     }))
   }
 
-  public recordReorderUndone(): Promise<void> {
+  private recordReorderUndone(): Promise<void> {
     return this.updateDailyMeasures(m => ({
       reorderUndoneCount: m.reorderUndoneCount + 1,
     }))
   }
 
+  private recordSquashConflictsEncountered(): Promise<void> {
+    return this.updateDailyMeasures(m => ({
+      squashConflictsEncounteredCount: m.squashConflictsEncounteredCount + 1,
+    }))
+  }
+
+  public recordSquashMultipleCommitsInvoked(): Promise<void> {
+    return this.updateDailyMeasures(m => ({
+      squashMultipleCommitsInvokedCount:
+        m.squashMultipleCommitsInvokedCount + 1,
+    }))
+  }
+
+  private recordSquashSuccessful(): Promise<void> {
+    return this.updateDailyMeasures(m => ({
+      squashSuccessfulCount: m.squashSuccessfulCount + 1,
+    }))
+  }
+
+  private recordSquashSuccessfulWithConflicts(): Promise<void> {
+    return this.updateDailyMeasures(m => ({
+      squashSuccessfulWithConflictsCount:
+        m.squashSuccessfulWithConflictsCount + 1,
+    }))
+  }
+
+  public recordSquashViaContextMenuInvoked(): Promise<void> {
+    return this.updateDailyMeasures(m => ({
+      squashViaContextMenuInvokedCount: m.squashViaContextMenuInvokedCount + 1,
+    }))
+  }
+
+  public recordSquashViaDragAndDropInvokedCount(): Promise<void> {
+    return this.updateDailyMeasures(m => ({
+      squashViaDragAndDropInvokedCount: m.squashViaDragAndDropInvokedCount + 1,
+    }))
+  }
+
+  private recordSquashUndone(): Promise<void> {
+    return this.updateDailyMeasures(m => ({
+      squashUndoneCount: m.squashUndoneCount + 1,
+    }))
+  }
+
+  public recordOperationConflictsEncounteredCount(
+    kind: MultiCommitOperationKind
+  ): Promise<void> {
+    switch (kind) {
+      case MultiCommitOperationKind.Squash:
+        return this.recordSquashConflictsEncountered()
+      case MultiCommitOperationKind.Reorder:
+        return this.recordReorderConflictsEncountered()
```

Comment:
Just throwing an idea… I think it's not compile-time safe (or maybe it is with the proper TypeScript knowledge 😂 ), but maybe we can leverage the string values behind the `MultiCommitOperationKind` to compose the key we want to update in the daily measures structure.

Something like…
```
const key = `${kind.toLowerCase()}ConflictsEncounteredCount`
return this.updateDailyMeasures(m => ({
      `${key}`: m[key] + 1,
}))
```

Reply:
:D - I thought of that.. but I have always been told it is kind of bad practice since you are kind skipping typing..